### PR TITLE
Avoid retaining self in block

### PR DESCRIPTION
**Pull request by dnfield**
Speculatively fixes https://github.com/flutter/flutter/issues/45592

Need to figure out a proper test for this.

/cc @PJLX

### CODE REVIEW

File: shell/platform/darwin/ios/framework/Source/FlutterEngine.mm [hunk 1]
```diff
@@ -295,9 +295,13 @@ - (void)resetChannels {
 - (void)setupChannels {
   // This will be invoked once the shell is done setting up and the isolate ID
   // for the UI isolate is available.
+  __block fml::WeakPtr<FlutterEngine> weakSelf = [self getWeakPtr];
```
Comment (chinmaygarde) [inline shell/platform/darwin/ios/framework/Source/FlutterEngine.mm]: The __block specified does nothing for fml::WeakPtr.

**Comment by jmagman** [comment]
We don't need to reflexively prevent self from being retained in blocks, as long as it's on purpose and done thoughtfully.  If the binaryMessageHandler needs the engine to stay alive, that's fine, as long as something releases the binaryMessageHandler and its block well before the engine is expected to be released.

File: shell/platform/darwin/ios/framework/Source/FlutterEngine.mm (hunk 1 above)
Comment (dnfield) [reply shell/platform/darwin/ios/framework/Source/FlutterEngine.mm]: Done.

**Comment by dnfield** [comment]
@jmagman the binary messenger is designed to not retain the engine - it asks for the pointer and then does nil checks later. I had probably incorrectly assumed that this case should never arise.

**Pull request merged**